source/extensions/filters/common/rbac/matchers.cc

```diff
@@ -123,10 +128,21 @@ bool HeaderMatcher::matches(const Network::Connection&,
 }
 
 bool IPMatcher::matches(const Network::Connection& connection, const Envoy::Http::RequestHeaderMap&,
-                        const StreamInfo::StreamInfo&) const {
-  const Envoy::Network::Address::InstanceConstSharedPtr& ip =
-      destination_ ? connection.localAddress() : connection.remoteAddress();
-
+                        const StreamInfo::StreamInfo& info) const {
+  Envoy::Network::Address::InstanceConstSharedPtr ip;
+  switch (type_) {
+  case ConnectionLocal:
+    ip = connection.localAddress();
+    break;
+  case ConnectionRemote:
+    ip = connection.remoteAddress();
```

Comment:
change to `info.downstreamDirectRemoteAddress()`?

Reply:
So I assume it's always the same as the address from connection?

Reply:
Yes, except there may be cases where it is not set. It is currently set for connection-level streamInfo (formerly tcp_proxy), dubbo proxy streams, thrift proxy streams, and http streams. This covers most things, but I don't know exactly how RBAC is used/configured.

Reply:
The RBAC filter is used either as a network filter or HTTP filter inside the HttpConnectionManager, I'm not sure how is this related to specific protocol (e.g. dubbo, thrift, etc.) but I assume it will always return non null in these use cases?

Reply:
Yes, it will be non-null in both of those cases.